src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2919,18 +2924,66 @@ public bool HasBoxingConversion(TypeSymbol source, TypeSymbol destination, ref H
             return false;
         }
 
-        internal static bool HasImplicitPointerConversion(TypeSymbol source, TypeSymbol destination)
+        internal static bool HasImplicitPointerToVoidConversion(TypeSymbol source, TypeSymbol destination)
         {
             Debug.Assert((object)source != null);
             Debug.Assert((object)destination != null);
 
             // SPEC: The set of implicit conversions is extended to include...
             // SPEC: ... from any pointer type to the type void*.
 
-            PointerTypeSymbol pd = destination as PointerTypeSymbol;
-            PointerTypeSymbol ps = source as PointerTypeSymbol;
-            return (object)pd != null && (object)ps != null && pd.PointedAtType.IsVoidType();
+            return source.IsPointerOrFunctionPointer() && destination is PointerTypeSymbol { PointedAtType: { SpecialType: SpecialType.System_Void } };
+        }
+
+#nullable enable
+        internal bool HasImplicitPointerConversion(TypeSymbol? source, TypeSymbol? destination, ref HashSet<DiagnosticInfo>? useSiteDiagnostics)
```

Comment:
>HasImplicitPointerConversion [](start = 22, length = 28)

It looks like this function only checks for function pointer conversions. Why don't we reflect that in its name and in the corresponding ConversionKind name?

Reply:
@gafter suggested that we make this broad enough to encompass any future implicit conversions we want to support.